pkg/controller/build/build_controller.go

```diff
@@ -312,48 +315,57 @@ func (r *ReconcileBuild) validateSecrets(ctx context.Context, secretNames []stri
 	return nil
 }
 
-func (r *ReconcileBuild) configFinalizer(ctx context.Context, b *build.Build) error {
-	if b.GetAnnotations()[build.AnnotationBuildRunDeletion] == "true" {
-		if !contains(b.GetFinalizers(), build.BuildFinalizer) {
-			ctxlog.Info(ctx, "add finalizer to build", namespace, b.Namespace, name, b.Name)
-			b.SetFinalizers(append(b.GetFinalizers(), build.BuildFinalizer))
-			if err := r.client.Update(ctx, b); err != nil {
-				return err
+// validateBuildRunOwnerReferences defines or removes the ownerReference for the BuildRun based on
+// an annotation value
+func (r *ReconcileBuild) validateBuildRunOwnerReferences(ctx context.Context, b *build.Build) error {
+	buildRunList, err := r.retrieveBuildRunsfromBuild(ctx, b)
+	if err != nil {
+		return err
+	}
+
+	if b.GetAnnotations()[build.AnnotationBuildRunDeletion] == "false" {
+		// if the buildRun have an ownerreference to the Build, lets remove it
+		for _, buildRun := range buildRunList.Items {
+			if index := r.validateBuildOwnerReference(buildRun.OwnerReferences, b); index != -1 {
+				buildRun.OwnerReferences = removeOwnerReferenceByIndex(buildRun.OwnerReferences, index)
+				if err := r.client.Update(ctx, &buildRun); err != nil {
+					return err
+				}
```

Comment:
There is a log statement in the `else` branch which is not here:

`ctxlog.Info(ctx, fmt.Sprintf("succesfully updated BuildRun %s", buildRun.Name), namespace, buildRun.Namespace, name, buildRun.Name)`

Reply:
good catch, done, thanks